src/simulation.hpp

```diff
@@ -2374,35 +2380,76 @@ template <typename problem_t> auto AMRSimulation<problem_t>::PlotFileMFAtLevel(c
 
 	// copy data from cell-centred state variables
 	for (int i = 0; i < ncomp_cc; i++) {
-		amrex::MultiFab::Copy(plotMF, state_new_cc_[lev], i, comp, 1, included_ghosts);
+		amrex::MultiFab::Copy(plotMF_cc, state_new_cc_[lev], i, comp, 1, included_ghosts);
 		comp++;
 	}
 
 	// compute cell-center averaged face-centred data
 	if constexpr (Physics_Indices<problem_t>::nvarTotal_fc > 0) {
 		for (int idim = 0; idim < AMREX_SPACEDIM; ++idim) {
-			AverageFCToCC(plotMF, state_new_fc_[lev][idim], idim, comp, 0, ncomp_per_dim_fc);
+			AverageFCToCC(plotMF_cc, state_new_fc_[lev][idim], idim, comp, 0, ncomp_per_dim_fc);
 			comp += ncomp_per_dim_fc;
 		}
 	}
 
 	// compute derived vars
 	for (auto const &dname : derivedNames_) {
-		ComputeDerivedVar(lev, dname, plotMF, comp);
+		ComputeDerivedVar(lev, dname, plotMF_cc, comp);
 		comp++;
 	}
 
-	return plotMF;
+	return plotMF_cc;
+}
+
+template <typename problem_t>
+auto AMRSimulation<problem_t>::PlotFileMFAtLevel_fc(const int lev, int idim, const int nghost_fc_) -> amrex::MultiFab // make fc version, refer to notes
+{
+	// get number of variables, includes framework for adding derived vars in future
+	int comp = 0;
+	int nvar_dim_tot_fc = 0;
+	if constexpr (Physics_Indices<problem_t>::nvarPerDim_fc > 0) {
+		nvar_dim_tot_fc = Physics_Indices<problem_t>::nvarPerDim_fc;
+	}
+	// const int ncomp_deriv = derivedNames_.size();
+	const int ncomp_plotMF_fc = nvar_dim_tot_fc;
+
+	amrex::MultiFab plotMF_fc(grids[lev], dmap[lev], ncomp_plotMF_fc, nghost_fc_);
+
+	// Fill ghost zones for state_new_fc_
+	if constexpr (Physics_Indices<problem_t>::nvarPerDim_fc > 0) {
+		fillBoundaryConditions(state_new_fc_[lev][idim], state_new_fc_[lev][idim], lev, tNew_[lev], quokka::centering::fc,
+				       static_cast<quokka::direction>(idim), InterpHookNone, InterpHookNone, FillPatchType::fillpatch_function);
+	}
+
+	// copy data from face-centred state variables
+	for (int i = 0; i < ncomp_plotMF_fc; i++) {
+		amrex::MultiFab::Copy(plotMF_fc, state_new_fc_[lev][idim], i, comp, 1, nghost_fc_);
+		comp++;
+	}
+	return plotMF_fc;
 }
 
 // put together an array of multifabs for writing
-template <typename problem_t> auto AMRSimulation<problem_t>::PlotFileMF(const int included_ghosts) -> amrex::Vector<amrex::MultiFab>
+template <typename problem_t> auto AMRSimulation<problem_t>::PlotFileMF_cc(const int included_ghosts) -> amrex::Vector<amrex::MultiFab> // make fc version
 {
-	amrex::Vector<amrex::MultiFab> r;
+	amrex::Vector<amrex::MultiFab> r_cc;
```

Comment:
It's not necessary to change the names of variables in existing code that has no functional changes in this PR.